pkg/controllers/controllers.go

```diff
@@ -114,6 +114,7 @@ func Initialize(injectCloudProvider func(context.Context, cloudprovider.Options)
 		Scheme:                     scheme,
 		MetricsBindAddress:         fmt.Sprintf(":%d", opts.MetricsPort),
 		HealthProbeBindAddress:     fmt.Sprintf(":%d", opts.HealthProbePort),
+		BaseContext:                func() context.Context { return ctx },
```

Comment:
When I had looked into this previously, I wrote this code:
```go
BaseContext:            newRunnableContext(config, opts, logging.FromContext(ctx))
```
```go
func newRunnableContext(config *rest.Config, options options.Options, logger zap.SugaredLogger) func() context.Context {
       return func() context.Context {
               ctx := context.Background()
               ctx = logging.WithLogger(ctx, logger)
               ctx = injection.WithConfig(ctx, config)
               ctx = injection.WithOptions(ctx, options)
               return ctx
       }
}
```

I don't think it matters at the moment, but it seems to be safer for the BaseContext func to return new contexts.

Reply:
Can you explain why you think this is safer? I'm failing to think of issues with passing in the same context object into the func

Reply:
It's just that BaseContext is a function and not just a context and that the default implementation returns a new context via `context.Background()` for each call. Combining that with the behavior of the cancels cancelling child contexts:
```go
// The WithCancel, WithDeadline, and WithTimeout functions take a
// Context (the parent) and return a derived Context (the child) and a
// CancelFunc. Calling the CancelFunc cancels the child and its
// children, removes the parent's reference to the child, and stops
// any associated timers. Failing to call the CancelFunc leaks the
// child and its children until the parent is canceled or the timer
// fires. The go vet tool checks that CancelFuncs are used on all
// control-flow paths.
```

A cancel on that top-level context would cancel all of the runner's contexts.  Maybe that's ok? It seemed safer to use new contexts than figure that out.

Reply:
You're completely correct. Since they are just calling the `baseContext` function to get the parent context for the `context.WithCancel` used by the manager runner